libcxx/include/__format/formatter_integral.h

```diff
@@ -148,14 +151,15 @@ _LIBCPP_HIDE_FROM_ABI auto __format_char(
 // Integer
 //
 
-/** Wrapper around @ref to_chars, returning the output pointer. */
-template <integral _Tp>
-_LIBCPP_HIDE_FROM_ABI char* __to_buffer(char* __first, char* __last, _Tp __value, int __base) {
+/** Wrapper around @ref to_chars, returning the output iterator. */
+template <contiguous_iterator _Iterator, integral _Tp>
+_LIBCPP_HIDE_FROM_ABI _Iterator __to_buffer(_Iterator __first, _Iterator __last, _Tp __value, int __base) {
```

Comment:
Could we bubble the changes to the point of use since this is an internal function? That would limit the number of instantiations this function (and other detail functions) ends up generating.

Reply:
I think this comment is the same as Mark's comment above (https://github.com/llvm/llvm-project/pull/74624#discussion_r1417654689)? If not, I misunderstood your comment.

Reply:
Ah, yep, let's continue there.